10up/ElasticPress
# Migrate Autosuggest filters to proper WordPress hooks

### Description of the Change
Migrates the custom 'filter' functions in Autosuggest to use proper WordPress hooks. Existing uses of filter functions are applied as callbacks to the new hooks from a back-compat file.


Closes #2961

### How to test the Change
- Add the example code from the _existing_ documentation and confirm they still behave as expected.
- Replace the code with the example code from the updated documentation and confirm the result still matches what is described.

### Changelog Entry
- Added `ep.Autosuggest.itemHTML`, `ep.Autosuggest.listHTML`, `ep.Autosuggest.query`, and `ep.Autosuggest.element` JavaScript hooks to Autosuggest and migrated filter functions to hook callbacks for backwards compatibility.

### Credits

Props @JakePT

### Checklist:
